config-model/src/main/java/io/strimzi/kafka/config/model/generator/KafkaConfigModelGenerator.java

```diff
@@ -2,7 +2,7 @@
  * Copyright Strimzi authors.
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
-package io.strimzi.build.kafka.metadata;
+package io.strimzi.kafka.config.model.generator;
```

Comment:
This means that the generator will be present in the classpath at runtime, which wasn't the case before. I think this means that kafka itself appears to be a (transitive) runtime dependency of the operator. I think we want to avoid that, if possible, because it means someone will end up using that dependency in operator code. We don't want that kind of coupling.

Reply:
Sorry not sure I follow exactly what do you mean with this. Can you elaborate a bit more on this and explain ehat issues do you expect from it?

Reply:
I mean, yes, this moves the generator to runtime. The rest is not completely clear to me.

Reply:
Previously Kafka was not a dependency of the operator. Changing that allows operator code to make use of Kafka APIs, and it would be easy to miss in code review. Making use of Kafka APIs risks operator behaviour becoming coupled to some particular _version_ of Kafka (which then could mean weird incompatiblities in the operator, depending on when the dependency was upgraded). It would also make it easier for the operator to depend on internal Kafka APIs (again, easily missed in code review). Avoiding these kinds of things was the reason we came up with the config model in the first place. Otherwise we could have just had the operator depend directly on Kafka's `KafkaConfig` and `Validator` classes, which would be much simpler, but prone so subtle bugs or compatibility problems if those changed between Kafka versions.

Reply:
Ok, understand now. But, well, the previous model was broken - I would argue that it was broken way more than introducing the Kafka dependency. 

So what other alternatives do you see? The previous iteration of this had the modules separate, but that requires you to always to `mvn install` instead of `mvn test` or `mvn verify` for example. I can probably live with it and it is better than the current state. But it was not perfect either.

Reply:
I think keeping the modules separate would be preferable. I hear your build pain though. I think the right way to solve that would be to factor the generator into something (maybe a custom maven build plugin?) which the operator build could simply depend on (not have to build). But that's pretty costly, since it would require a release and push to maven central. Or maybe scripting.

Reply:
Ok, well, I pushed the alternative implementation with separate modules then. Please have a look.